backpack/__init__.py

```diff
@@ -205,7 +213,7 @@ def hook_run_extensions(
         print("[DEBUG] Running extension hook on", module)
     CTX.get_extension_hook()(module)
 
-    if not (
+    if not CTX.get_retain_graph() and not (
```

Comment:
write `not a and not b` as `not (a or b)`

Reply:
Done.